minio/minio
# Disable browser redirect

## Description
Introduce a new environment variable to disable redirecting to the Minio browser for files that don't exist or you don't have access to.

## Motivation and Context
When trying to access an object with insufficient rights or an object that does not exist from a web browser, it can be undesired to be redirected to the Minio web interface. Using the `MINIO_BROWSER_REDIRECT` environment variable, it is possible to disable this behaviour, returning a regular 403 (XML) response like you would expect from Amazon S3 as well.

Alternatively, it would be nice to have the option to bind to a second port which (only) exposes the Minio browser, while the regular listen port serves regular requests (without redirecting). That way, it would be possible to add extra security checks when running behind a reverse proxy.

## How to test this PR?
1. Build the minio server and start it with the `MINIO_BROWSER_REDIRECT` set to `off`.
```
make build
MINIO_BROWSER_REDIRECT=off ./minio server /tmp/minio-data
```
2. Try to open a random file in your web browser: `http://127.0.0.1:9000/non-existing-bucket/filename.example`
3. Expect to see a 403 response saying `AccessDenied` instead of being redirected to `http://127.0.0.1:9000/minio/non-existing-bucket/filename.example`
4. Verify that the browser still works by visiting `http://127.0.0.1:9000/minio/`
5. Restart the server with `MINIO_BROWSER_REDIRECT` set to `on`.
```
MINIO_BROWSER_REDIRECT=off ./minio server /tmp/minio-data
```
6. Verify that the previous URL now does redirect to the Minio browser.

## Types of changes

## Checklist:
